generate_workspace/src/test/java/com/google/devtools/build/workspace/output/BzlWriterTest.java

```diff
@@ -35,18 +42,33 @@
   @Test
   public void writeEmpty() throws Exception {
     BzlWriter writer = new BzlWriter(new String[]{}, System.getenv("TEST_TMPDIR"));
-    writer.write(ImmutableList.of());
+    writer.write(createRules());
     String fileContents = Files.toString(
         new File(System.getenv("TEST_TMPDIR") + "/generate_workspace.bzl"),
         Charset.defaultCharset());
     assertThat(fileContents).contains("def generated_maven_jars():\n  pass\n");
     assertThat(fileContents).contains("def generated_java_libraries():\n  pass\n");
   }
 
+  /** Ensures that it automatically creates output directory if they do not exist */
+  @Test
+  public void automaticallyCreateParentDirectories() throws Exception {
+    FileSystem fs = Jimfs.newFileSystem(Configuration.unix());
+    Path root = fs.getPath("root");
+    createDirectories(root);
+
+    Path outputDir = root.resolve("child/directory/");
+
+    BzlWriter writer = new BzlWriter(new String[]{}, outputDir.toString());
```

Comment:
But you're using an in-memory filesystem in the test and BzlWriter will try to write to /root/child/directory (IIUC).